go-controller/pkg/allocator/udn/allocator.go

```diff
@@ -0,0 +1,94 @@
+package udn
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type Allocator struct {
+	watchFactory factory.NodeWatchFactory
+	nodeName     string
+}
+
+func New(watchFactory factory.NodeWatchFactory, nodeName string) *Allocator {
+	return &Allocator{
+		watchFactory: watchFactory,
+		nodeName:     nodeName,
+	}
+}
+
+func (a *Allocator) AllocateConntrackMark(networkName string) (uint, error) {
+	return a.calculateIDsFromNetwork("ct-mark", networkName, config.Gateway.UserDefinedNetworkConntrackMarkBase, config.Default.MaxUserDefinedNetworks, 1)
+}
+
+func (a *Allocator) AllocateVRFTable(networkName string) (uint, error) {
+	return a.calculateIDsFromNetwork("vrf-table", networkName, config.Gateway.UserDefinedNetworkVRFTableBase, config.Default.MaxUserDefinedNetworks, 1)
+}
+
+func (a *Allocator) AllocateV4MasqueradeIPs(networkName string) ([]net.IP, error) {
+	return a.allocateMasqueradeIPs(networkName, "v4", config.Gateway.V4MasqueradeSubnet)
+}
+
+func (a *Allocator) AllocateV6MasqueradeIPs(networkName string) ([]net.IP, error) {
+	return a.allocateMasqueradeIPs(networkName, "v6", config.Gateway.V6MasqueradeSubnet)
+}
+
+func (a *Allocator) allocateMasqueradeIPs(networkName, idPrefix, masqueradeSubnet string) ([]net.IP, error) {
```

Comment:
@qinqon for my education, it returns list of Masquerade IPs, which IP to use for pod2egress SNAT ?

Reply:
@pperiyasamy this is returning a pair of them, whoever implement pod2egress should choose wich one if for what, so whatever you see fit.